### PR TITLE
Qt/Input: Introduce profiles

### PR DESCRIPTION
**Pull request by Megamouse**
This also fixes a XInput connection issue, where a disconnected device would incapacitate all following devices.

You can now add profiles by just putting them into the correct folder or creating them yourselves with the corresponding button.
please test this like bosses minna

confirmations:
- [x] keyboard
- [x] Xinput
- [x] DS4
- [x] MMJOY
- [x] evdev

### CODE REVIEW

**Comment by Asinin3** [comment]
If i chose a profile e.g profile 4 for my pad1. after hitting "ok" and closing the pad window it resets back to profile 0. Making it impossible to use more than one profile.

**Comment by Megamouse** [comment]
@Asinin3 it should work now

**Comment by toccata10** [comment]
A welcomed functionality. About the ergonomy: I would prefer these 2 things:
1) display the evdev number next (before or after) to the controller name on the gamepad settings panel
2) be able to load/save a file instead of a profile number as I can give more explicit names. This probably means duplicate the file selection on the "configure evdev" panel and add a "load" button on it.

**Comment by MSuih** [comment]
Woudn't it be better to store the profiles as a vector rather than a fixed-lenght array? That way you could arrange them more freely and delete unneeded profiles from the middle of the list. You'd have to add profile naming though, else it would get confusing really fast. Some kind of dynamic loading would be worth looking into as well to avoid clogging up the system if users create large amounts of profiles.

Saving the profiles as individual files and then f.ex. loading a whole /profiles folder into the UI would be an neat feature as well, but I'd imagine it woudn't be easy to implement since you'd have to work through the OS and it's file system.

**Comment by MrHulaHoop** [comment]
Profiles on keyboard work. Tested with Catherine Co-op using different profiles for each player on the same keyboard. 
If I had 2 keyboards connected, I imagine binding separate profiles to different keyboards shouldn't be an issue either.

**Comment by Megamouse** [comment]
reworked to use custom profiles, each an own file in /InputConfigs/<handler>/<profile>.yml

should also have an experimental evdev event nr. (no idea if that works)

**Comment by Megamouse** [comment]
@toccata10 can you please test this with evdev again?

**Comment by toccata10** [comment]
It works nicely with evdev.
1) Now, I wonder why "evdev_positive_axis" is a separate file: shouldn't the parameters be included at the end of the yml profile file ? I wondered why after changing ABS_X and ABS_Y to true, it works fine with my dragonrise and xbox360 sticks (I predicted it shouldn't work with xbox360, but it works).
2) the evdev event number works fine. That's good: it's easier to distinguish same brand controllers.
3) I suggest adding a "Delete profile" button: it'll be easier for everyone, instead of having to digg in the subfolders.
4) I should get my hands on a DS4 controller soon, so I'll can test that.
Very cool feature being able to remap and mix different controllers.

**Comment by Megamouse** [comment]
Thank you for testing!

1. this PR doesn't handle the positive axis for multiple controllers yet.
2. Good to know, I was just changing two lines of code without thinking. I'm kinda surprised that it worked.
3. The delete profile button will not be part of this PR (yet)
4. yeah.. DS4 would be nice, but don't buy it just for this purpose.. unless you wanted to have one anyway

**Comment by toccata10** [comment]
The DS4 was on its way (you know, Christmas time...), but sadly it segfaults as soon as I select it (and closes rpcs3):
selection_057
On master, it works fine (even the gyros)

**Comment by Megamouse** [comment]
could you try to find the reason?

**Comment by toccata10** [comment]
That seems way beyond my skills. I'll see if I can create a backtrace.

**Comment by toccata10** [comment]
My backtrace:
https://pastebin.com/m3pMkeLQ
Doesn't to like this: "ds4_pad_handler::SendVibrateData"

**Comment by Megamouse** [comment]
I changed sth.
could you please test again?
and if it works: please try the vibration tests in the pad-dialog

**Comment by toccata10** [comment]
ds4 is working fine now. I tried the switch in the pad-dialog: works fine. In game vibrations: working.
So: this profile stuff is working fine on my side. Another nice addition.
On Super Street Fighter Turbo 2 HD (and 1942 joint strike), the buttons are not responding but I suppose it's in your other PR #3989

File: rpcs3/Emu/Io/PadHandler.h [hunk 1]
```diff
@@ -314,6 +359,8 @@ struct pad_config : cfg::node
 	}
 };
 
+static input_config input_cfg;
```
Comment (Nekotekina) [inline rpcs3/Emu/Io/PadHandler.h]: What is this? Don't declare such variables in headers.

**Pull request merged**